Orange/widgets/utils/plot/owplotgui.py

```diff
@@ -27,18 +27,225 @@
+        super().closeEvent(QCloseEvent)
+
+class VariableSelection(QObject):
+    def __init__(self, master):
+        super().__init__()
+        self.master = master
+
+    def __call__(self):
+        master = self.master
+
+        params_view = {"sizePolicy": QSizePolicy(*SIZE_POLICY_ADAPTING),
+                       "selectionMode": QListView.ExtendedSelection,
+                       "dragEnabled": True,
+                       "defaultDropAction": Qt.MoveAction,
+                       "dragDropOverwriteMode": False,
+                       "dragDropMode": QListView.DragDrop}
+
+        self.active_view = view = gui.listView(widget=master.controlArea, master=master,
+                                               box="Displayed Axes", **params_view)
+
+        view.box.setMinimumHeight(120)
+        view.viewport().setAcceptDrops(True)
+
+        movedown = QAction(
+            "Move down", view,
+            shortcut=QKeySequence(Qt.Key_Delete),
+            triggered=self.__deactivate_selection
+        )
+        view.addAction(movedown)
+
+        order = (DomainModel.METAS, DomainModel.ATTRIBUTES)
+        model_params = dict(
+            parent=master, enable_dnd=True, order=order, valid_types=ContinuousVariable)
+        self.varmodel_selected = model = DomainModel(**model_params)
+
+        model.rowsInserted.connect(master._invalidate_plot)
+        model.rowsRemoved.connect(master._invalidate_plot)
+        model.rowsMoved.connect(master._invalidate_plot)
+
+        view.setModel(model)
+
+        addClassLabel = QAction("+", master,
+                                toolTip="Add new class label",
+                                triggered=self._action_add)
+        removeClassLabel = QAction(unicodedata.lookup("MINUS SIGN"), master,
+                                   toolTip="Remove selected class label",
+                                   triggered=self.__deactivate_selection)
+
+        add_remove = itemmodels.ModelActionsWidget([addClassLabel, removeClassLabel], master)
+        add_remove.layout().addStretch(10)
+        add_remove.layout().setSpacing(1)
+        add_remove.setSizePolicy(*SIZE_POLICY_FIXED)
+        master.controlArea.layout().addWidget(add_remove)
+        self.add_remove = add_remove
+        self.box = add_remove.buttons[1]
+
+        self.varmodel_other = model = DomainModel(**model_params)
+
+        self.other_view = view = gui.listView(None, master, box="Hidden Axes", **params_view)
+        view.setMinimumSize(QSize(30, 60))
+        view.setSizePolicy(*SIZE_POLICY_ADAPTING)
+        view.viewport().setAcceptDrops(True)
+        moveup = QAction(
+            "Move up", view,
+            shortcut=QKeySequence(Qt.Key_Delete),
+            triggered=self.__activate_selection
+        )
+        view.addAction(moveup)
+
+        view.setModel(model)
+        self.filter_edit, self.other_view = variables_filter(
+            model=model)
+        self.hidden = Hidden(self)
```

Comment:
I believe that a huge part of the mess related to models comes from having `Hidden` as a persistent dialog. Make it a normal modal dialog that is created when needed, and simply initialized with all attributes appearing in the domain that are currently hidden.